benchmarks/mmmu/mmmu.py

```diff
@@ -0,0 +1,169 @@
+import ast
+import base64
+from io import BytesIO
+
+from datasets import concatenate_datasets, load_dataset
+from PIL import Image
+
+from inspect_ai import Task, task
+from inspect_ai.dataset import Sample
+from inspect_ai.dataset._dataset import MemoryDataset
+from inspect_ai.model import ChatMessageUser, ContentImage, ContentText, GenerateConfig
+from inspect_ai.scorer import choice, model_graded_fact
+from inspect_ai.solver import generate, multiple_choice, prompt_template
+
+
+# Helper function to convert images to data URLs
+def image_to_data_url(image):
+    if isinstance(image, Image.Image):
+        buffered = BytesIO()
+        image.save(buffered, format="PNG")
+        image_bytes = buffered.getvalue()
+    else:
+        with open(image, "rb") as image_file:
+            image_bytes = image_file.read()
+
+    encoded_string = base64.b64encode(image_bytes).decode("utf-8")
```

Comment:
Note that you can include _both_ an image and text in a chat message (list of content where one item is of type image and the other of type text)

Reply:
I've changed how the images are added to the content list, to the following:

- Save images to disk, with unique file paths for every (question, image).
- After adding the question to the content list, I add the string `<image 1>`, followed by `<image 1>`'s file path. I repeat this for every image the question contains. I thought this would be a clean way of getting round the issue of OpenAI not allowing image metadata. Now, if an image is referred to in one of the multiple-choice answers, I think the model should have some information on which image is which. I've added some examples below.
![image](https://github.com/user-attachments/assets/8915257c-acaa-412f-abfc-a5e2f66b7216)
![image](https://github.com/user-attachments/assets/61b2abb7-9ac9-40cb-a8ff-3a05c7a9c75f)

Reply:
Minor note: I've changed the temporary file paths to be of the format `temp_images/Subject/question_i_image_j.png`, just for a bit more clarity.

Reply:
Update: 
- I've removed the above `<image n>` additions to the content list. I was looking into whether the order of the content changes for the API submission, and saw that it seems to, from the View API tab (screenshot below). 
- Performance is the same whether these tags are inserted or not. 
- I ran a quick test on the ChatGPT UI, and it seems GPT-4o is able to infer which image is which, from the order they're submitted (screenshot below). 
- As a result, I don't think we need any additional logic to deal with images referenced in the question or answer. 

![image](https://github.com/user-attachments/assets/f284a615-016d-4bc4-ba29-1df1fc898801)

![image](https://github.com/user-attachments/assets/f1ad43ee-b1b6-4f74-9bb7-a9487df44b6b)